src/ansys/geometry/core/designer/component.py

```diff
@@ -174,15 +175,24 @@ def __init__(
         if preexisting_id:
             self._name = name
             self._id = preexisting_id
+            self._intance_name = instance_name
         else:
+            print("no preexisting_id")
             if parent_component:
                 template_id = template.id if template else ""
                 new_component = self._component_stub.Create(
-                    CreateRequest(name=name, parent=parent_component.id, template=template_id)
+                    CreateRequest(
+                        name=name,
+                        parent=parent_component.id,
+                        template=template_id,
+                        instance_name=instance_name,
+                    )
                 )
+
                 # Remove this method call once we know Service sends correct ObjectPath id
                 self._id = new_component.component.id
                 self._name = new_component.component.name
+                self._intance_name = instance_name
             else:
                 self._name = name
                 self._id = None
```

Comment:
You should also initialize to None the instance name here
```suggestion
                self._id = None
                self._instance_name = None
```

Reply:
Do not resolve comments you have not committed @umutsoysalansys

Reply:
sorry I can't follow you really.